Sources/StreamChat/APIClient/Endpoints/Payloads/UserPayloads.swift

```diff
@@ -29,6 +29,8 @@ enum UserPayloadsCodingKeys: String, CodingKey, CaseIterable {
 
 /// An object describing the incoming user JSON payload.
 class UserPayload: Decodable {
+    static var userDecodingCache = [String: UserPayload]()
```

Comment:
this is not thread-safe

Reply:
Alright, I've switched to NSCache which is thread-safe:
> You can add, remove, and query items in the cache from different threads without having to lock the cache yourself.